pkg/services/ngalert/schedule/schedule.go

```diff
@@ -395,35 +382,24 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 	}
 
 	evalRunning := false
-	var currentRule *ngmodels.AlertRule
+	var currentRuleVersion int64 = math.MinInt64
 	var extraLabels map[string]string
 	defer sch.stopApplied(key)
 	for {
 		select {
 		// used by external services (API) to notify that rule is updated.
-		case version := <-updateCh:
+		case lastVersion := <-updateCh:
 			// sometimes it can happen when, for example, the rule evaluation took so long,
 			// and there were two concurrent messages in updateCh and evalCh, and the eval's one got processed first.
 			// therefore, at the time when message from updateCh is processed the current rule will have
 			// at least the same version (or greater) and the state created for the new version of the rule.
-			if currentRule != nil && int64(version) <= currentRule.Version {
-				logger.Info("skip updating rule because its current version is actual", "current_version", currentRule.Version, "new_version", version)
+			if currentRuleVersion >= int64(lastVersion) {
```

Comment:
We should make `updateCh` a chan of `int64`?

Reply:
I use an alias ruleVersion for int64 to better reflect the meaning of int64. The conversion is a noop here.